#! /usr/bin/env python

# Ocean optics dynamic parameter configuration file

PACKAGE='ocean_optics'

from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

# clamp integration time at 30000us, even though we can go much higher
gen.add("integration_time", int_t, 0, "Sensor integration time (us)", 1000, 1000, 30000)

gen.add("boxcar_width", int_t, 0, "Boxcar filter width", 0, 0, 10)

gen.add("scans_to_average", int_t, 0, "# of scans to average", 1, 1, 10)

gen.add("correct_elec_dark", bool_t, 0, "Correct for electric dark", True, False, True)

exit(gen.generate(PACKAGE, "ocean_optics", "OceanOptics"))
